src/_cffi_src/build_openssl.py

```diff
@@ -38,8 +38,15 @@ def _osx_libraries(build_static):
 
 
 _PRE_INCLUDE = """
+#include <openssl/opensslv.h>
+/*
+    LibreSSL removed e_os2.h from the public headers so we'll only include it
+    if we're using vanilla OpenSSL.
+*/
+#if !defined(LIBRESSL_VERSION_NUMBER)
 #include <openssl/e_os2.h>
```

Comment:
But LibreSSL on windows is fine? (lol, no one is using that)

On Mon, Mar 28, 2016 at 10:15 PM, Paul Kehrer <notifications@github.com>
wrote:

> In src/_cffi_src/build_openssl.py
> <https://github.com/pyca/cryptography/pull/2861#discussion_r57664111>:
>
> > @@ -38,8 +38,15 @@ def _osx_libraries(build_static):
> >
> >
> >  _PRE_INCLUDE = """
> > +#include <openssl/opensslv.h>
> > +/*
> > +    LibreSSL removed e_os2.h from the public headers so we'll only include it
> > +    if we're using vanilla OpenSSL.
> > +*/
> > +#if !defined(LIBRESSL_VERSION_NUMBER)
> >  #include <openssl/e_os2.h>
>
> Yeah, that include defines a bunch of things that are used within other
> header files in openssl. Removing it will cause us issues on Windows.
>
> —
> You are receiving this because you commented.
> Reply to this email directly or view it on GitHub
> <https://github.com/pyca/cryptography/pull/2861/files/2e10ea4650c3c9ff0ce6881083c84945326122c8#r57664111>
>



-- 
"I disapprove of what you say, but I will defend to the death your right to
say it." -- Evelyn Beatrice Hall (summarizing Voltaire)
"The people's good is the highest law." -- Cicero
GPG Key fingerprint: D1B3 ADC0 E023 8CA6